slatedb/src/compactor_executor.rs

```diff
@@ -32,26 +32,26 @@ use log::{debug, error};
 use tracing::instrument;
 use ulid::Ulid;
 
-/// Execution unit (attempt) for a compaction plan.
+/// Arguments for starting a compaction job.
 ///
-/// - `id` is the job id (ULID) and uniquely identifies a single execution attempt. This is
+/// - `id` is the job id (ULID) and uniquely identifies a single job. This is
 ///   used as the runtime key in `scheduled_compactions`.
-/// - `job_id` is the canonical plan id (ULID) that ties this job attempt back to its
-///   `CompactionJob` entry in the compactor's canonical map.
+/// - `job_id` is the canonical plan id (ULID) that ties this job back to its
+///   `Compaction`
 ///
 /// Jobs carry fully materialized inputs (L0 `ssts` and `sorted_runs`) along with execution-time
 /// metadata for progress reporting, retention, and resume logic.
 #[derive(Clone, PartialEq)]
-pub(crate) struct CompactorJobAttempt {
-    /// Job attempt id. Unique per attempt and used for scheduling/routing.
+pub(crate) struct StartCompactionJobArgs {
```

Comment:
I thought this was going to become a `CompactionJobSpec` since it's a spec for a compaction job (an execution of a compaction)?

Reply:
From @rodesai:

> Its not really a spec since it has IDs in it so I just punted it for this patch. I guess I could just pull out the non-id stuff into a spec type. But then I was thinking we reserve spec for what we put into the core compactor model, which we wouldn't do with this type. So for now we just need a type to capture the information passed between the compactor/executor